File: frontend/web/features/management/types/classroom.ts
export type ClassroomValues = Record<string, string> & {
  classroomCode: string;
  building: string;
  roomType: string;
  departmentId: string;
  department: string;
  capacity: string;
  status: string;
  studyStatus: string;
  deviceOnline: string;
  createAt: string;
};

export type ClassroomItem = { id: string; values: ClassroomValues };

File: backend/src/InstituteManagement.Domain/Entities/Classroom.cs
namespace InstituteManagement.Domain.Entities;

public sealed class Classroom : Entity
{
    public required string ClassroomCode { get; set; }
    public string Building { get; set; } = string.Empty;
    public string RoomType { get; set; } = "Classroom";
    public int Capacity { get; set; }
    public Guid? DepartmentId { get; set; }
    public Department? Department { get; set; }
    public string Status { get; set; } = "Available";
    public bool DeviceOnline { get; set; } = true;
}
